Sources/SwiftCliCore/ChartConstructor.swift

```diff
@@ -3,40 +3,16 @@ import Foundation
 public class ChartConstructor {
     public init() {}
     public func makeChart(stitchArray: [[String]]) -> String {
-
-        let numberOfRows = stitchArray.count
+        let lastRow = stitchArray.count - 1
+        var patternMetaData = MetaDataBuilder().gatherAllMetaData(stitchArray: stitchArray)
         var finishedChart = ""
-        let lastRow = numberOfRows - 1
-
         for row in 0...lastRow {
-            let rowLength = stitchArray[row].count
-
-            if row == 0 {
-                finishedChart =  makeBottomRow(width: rowLength) + finishedChart
-                finishedChart = makeStitchRow(row: stitchArray[row]) + finishedChart
-            } else {
-                let prevRowLength = stitchArray[row - 1].count
-                let rightSideDiff = (rowLength - prevRowLength)
-                if rightSideDiff < 0 {
-                    let middleLine = makeMiddleRowStitchCountChange(width: rowLength, rDiff: rightSideDiff, lDiff: 0)
-                    finishedChart =  middleLine + finishedChart
-                } else {
-                    finishedChart = makeMiddleRow(width: rowLength) + finishedChart
-                }
-
-                finishedChart = makeStitchRow(row: stitchArray[row]) + finishedChart
-
-            }
-
-            if row == (numberOfRows - 1) {
-                finishedChart = makeTopRow(width: rowLength) + finishedChart
-
-            }
-
+            finishedChart = patternMetaData[row].totalRow + finishedChart
```

Comment:
Hm, couldn't you do something like:

```swift
reversedRows = patternMetaData.reversed()
finishedChart = reversedRows.map { $0.totalRow }
```

Then you also don't need to initialize `finishedChart` as an empty string first.

Reply:
It's less initializing the empty string that's the problem and more trying to reduce mutating variables (i.e. be in situations where you use `let` rather than `var`) as much as possible. The reason for that is because mutated state is harder to keep track of and introduces errors. You can join on an array. However, if it's taking too long to make this change work, let's put it on the back burner for now. 👍